rocketpy/motors/Motor.py

```diff
@@ -247,7 +379,11 @@ def totalImpulse(self):
         self.totalImpulse : float
             Motor total impulse in Ns.
         """
-        return self.thrust.integral(0, self.burnOutTime)
+        # Calculate total impulse
+        self.totalImpulse = self.thrust.integral(*self.burn_time)
```

Comment:
Just found out that since `self.totalImpulse` is a cached property, this line of code may go by without changing the cached value.

Here is a suggestion based on https://stackoverflow.com/questions/62662564/how-do-i-clear-the-cache-from-cached-property-decorator. Better solutions are welcome.
```suggestion
        # # Clear cache and calculate total impulse
        del self.totalImpulse 
        self.totalImpulse = self.thrust.integral(*self.burn_time)
```

Reply:
I prefer you change the cached property to a simple property then.
I have faced similar issue in the EnvironmentAnalysis class recently, and that was my approach.

Do not try to "clean" the cached property, if you need to update the value, just use property and we will be fine.

The `self.totalImpulse` is not even called a lot of times during the execution, so no worries.

Reply:
This is a great catch which turned out to be more significant that it seemed since the `reshapeThrustCurve` modified instance attributes that were cached, leading to incorrect values.

There is certainly more than one way to solve this issue, in [e4c75a1](https://github.com/RocketPy-Team/RocketPy/pull/337/commits/e4c75a158985fc72b1c23c73abf7a736a26c1262) I turned the `Motor` methods whose only function is to edit thrust values into `static`.

Therefore, the standard ways to reshape the thrust curve are:

- The same way as before, i.e., through the `__init__` parameter;
- Calling the static method as `Motor.reshapeThrustCurve` and producing a new thrust curve that may be used to instantiate a new motor.